static/src/javascripts/projects/common/modules/commercial/hosted-gallery.js

```diff
@@ -299,6 +301,8 @@ define([
                     bonzo(this.$galleryEl).toggleClass('show-oj', this.index === this.$images.length);
                     bonzo(this.$galleryEl).toggleClass('show-cta', this.index === config.page.ctaIndex + 1);
                 }
+
+                url.pushUrl({}, document.title, config.page.pageName + '#img-' + this.index, true);
```

Comment:
The pushUrl function does a replaceState if you pass true in the 4th param, so you don't have to press back 20 times. The window.location.hash thing would be simpler though..